### PR TITLE
Fix DRPolicy redirect issue after creation

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@SanjalKatiyar , @GowthamShanmugam Please review

File: packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx [hunk 1]
```diff
@@ -144,6 +141,15 @@ export const DRPolicyListPage: React.FC = () => {
     useModalLauncher(DRPolicyActions);
   const [isModalOpen, setConnectedAppsModalOpen] = React.useState(false);
   const [linkedApps, setLinkedApps] = React.useState<ApplicationRefKind[]>([]);
+
+  const acmMCOFlag = useFlag(ACM_OBSERVABILITY_FLAG);
+  const DATA_POLICIES_PATH = acmMCOFlag
+    ? '/multicloud/data-services/data-policies/recovery'
+    : '/multicloud/data-services/data-policies';
```
Comment (GowthamShanmugam) [inline packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx:148]: When acmMCOFlag is enabled but the user is non-admin, then the overview page won't be visible. So create DRPolicy page will redirect to the wrong link.

Comment (GowthamShanmugam) [reply packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx:148]: shall we use:
const location = useLocation();
const listPagePath: string = location.pathname;

cc: @SanjalKatiyar ?

Comment (TimothyAsirJeyasing) [reply packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx:148]: Ok, I will check both acmMCOFlag && adminFlag

File: packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx [hunk 2]
```diff
@@ -144,6 +141,17 @@ export const DRPolicyListPage: React.FC = () => {
     useModalLauncher(DRPolicyActions);
   const [isModalOpen, setConnectedAppsModalOpen] = React.useState(false);
   const [linkedApps, setLinkedApps] = React.useState<ApplicationRefKind[]>([]);
+
+  const acmMCOFlag = useFlag(ACM_OBSERVABILITY_FLAG);
+  const adminFlag = useFlag(ADMIN_FLAG);
```
Comment (GowthamShanmugam) [inline packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx:146]: try with:
const location = useLocation();
const listPagePath: string = location.pathname;

**Comment by TimothyAsirJeyasing** [comment]
/hold cancel
tested with 4 cases

File: packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx [hunk 3]
```diff
@@ -144,6 +139,13 @@ export const DRPolicyListPage: React.FC = () => {
     useModalLauncher(DRPolicyActions);
   const [isModalOpen, setConnectedAppsModalOpen] = React.useState(false);
   const [linkedApps, setLinkedApps] = React.useState<ApplicationRefKind[]>([]);
+
+  const location = useLocation();
+  const DATA_POLICIES_PATH: string = location.pathname.replace(/\/$/, '');
```
Comment (GowthamShanmugam) [inline packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx:144]:
```suggestion
  const drPolicyListPagePath: string = location.pathname.replace(/\/$/, '');
```

File: packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx [hunk 4]
```diff
@@ -144,6 +139,13 @@ export const DRPolicyListPage: React.FC = () => {
...
+  const DR_POLICY_CREATE_PATH = `${DATA_POLICIES_PATH}/${referenceForModel(
```
Comment (GowthamShanmugam) [inline packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx:145]:
```suggestion
  const drPolicyCreatePagePath = `${drPolicyListPagePath}/${referenceForModel(
```

**Comment by GowthamShanmugam** [comment]
1. Rebase with latest master
2. Run yarn format
3. Run yarn lint-fix
4. Run yarn i18n

File: packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx (hunk 4 above)
Comment (TimothyAsirJeyasing) [reply packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx]: Done, Thank you

File: packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx (hunk 3 above)
Comment (TimothyAsirJeyasing) [reply packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx]: Done

File: packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx [hunk 5]
```diff
@@ -144,6 +139,13 @@ export const DRPolicyListPage: React.FC = () => {
     useModalLauncher(DRPolicyActions);
   const [isModalOpen, setConnectedAppsModalOpen] = React.useState(false);
   const [linkedApps, setLinkedApps] = React.useState<ApplicationRefKind[]>([]);
+
+  const location = useLocation();
+  const drPolicyListPagePath: string = location.pathname.replace(/\/$/, '');
```
Comment (bipuladh) [inline packages/mco/components/disaster-recovery/drpolicy-list-page/drpolicy-list-page.tsx:144]:
```suggestion
  const drPolicyListPagePath = location.pathname.replace(/\/$/, '');
```
nit

**Comment by GowthamShanmugam** [comment]
This PR has some lint errors, please fix

**Comment by TimothyAsirJeyasing** [comment]
there is a lint error/warning is in master branch its appearing for every pr.

TopologyNavigation.tsx 42:5  warning  React Hook React.useCallback has a missing dependency: 'controller

**Review by GowthamShanmugam — APPROVED** [review]

**Pull request merged**